src/Tools/NodeDocumentationMarkdownGenerator/AssemblyHandler.cs

```diff
@@ -0,0 +1,387 @@
+                            }
+                        }
+
+                        else
+                        {
+                            Assembly asm = mlc.LoadFromAssemblyPath(path);
+                            AssemblyName name = asm.GetName();
+
+                            if (NodeModelAssemblyLoader.ContainsNodeModelSubTypeReflectionLoaded(asm, nodeModelType))
+                            {
+                                FileInfosFromNodeModels(asm, nodeModelType, ref nodeSearchModel);
+                                continue;
+                            }
+
+                            var dllDescriptors = GetFunctionDescriptorsFromDll(pathManager, asm);
+                            if (dllDescriptors != null)
+                            {
+                                functionDescriptors.AddRange(dllDescriptors);
+                            }
+                        }                  
+
+                        foreach (var descriptor in functionDescriptors)
+                        {
+                            Dictionary<string, FunctionGroup> fptrs;
+                            if (!functionGroups.TryGetValue(descriptor.Assembly, out fptrs))
+                            {
+                                fptrs = new Dictionary<string, FunctionGroup>();
+                                functionGroups[descriptor.Assembly] = fptrs;
+                            }
+
+                            string qualifiedName = descriptor.QualifiedName;
+                            FunctionGroup functionGroup;
+                            if (!fptrs.TryGetValue(qualifiedName, out functionGroup))
+                            {
+                                functionGroup = new FunctionGroup(qualifiedName);
+                                fptrs[qualifiedName] = functionGroup;
+                            }
+                            functionGroup.AddFunctionDescriptor(descriptor);
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        continue;
+                    }
+                }
+
+                if (functionGroups.Count > 0)
+                {
+                    AddZeroTouchNodesToSearch(functionGroups.Values.SelectMany(x => x.Values.Select(g => g)), ref nodeSearchModel);
+                }
+
+                fileInfos.AddRange(FileInfosFromSearchModel(nodeSearchModel));
+                Console.WriteLine($"{fileInfos.Count()} nodes found during scan");
+            }
+
+            return fileInfos;
+        }
+
+        private static List<FunctionDescriptor> GetFunctionDescriptorsFromDll(PathManager pathManager, Assembly asm)
+        {
+            var objectAsmName = typeof(object).Assembly.GetName().Name;
+            var descriptors = new List<FunctionDescriptor>();
+
+            string extension = System.IO.Path.GetExtension(asm.Location).ToLower();
+            if (extension != ".dll" && extension != ".exe")
```

Comment:
why is exe included?

Reply:
it shouldn't be? if the extension is anything else than `.dll` or `.exe` it gets excluded

Reply:
I'm confused - why do we need .exe, I understand .dll

Reply:
oh sorry, I think I misread your original question...

I followed the logic from the `ImportModuleHandler` which includes dll and exe, so didnt know if there are any scenarios where we need exe, happy to get rid of it just didnt know if there where any special cases that needed this https://github.com/DynamoDS/Dynamo/blob/master/src/Engine/ProtoCore/FFI/ImportModuleHandler.cs#L244

Reply:
ahh @aparajit-pratap any thoughts on why .exes can be imported?

Reply:
@mjkkirschner I can't think of an instance where we have imported exes in the past but I suppose it was included to import methods just like we do from dlls.